packages/kbn-config-schema/src/types/string_type.test.ts

```diff
@@ -130,6 +142,12 @@ describe('#validate', () => {
 
     expect(() => schema.string({ validate }).validate('foo')).toThrowErrorMatchingSnapshot();
   });
+
+  test('throw when empty string', () => {
+    const validate = () => 'validator failure';
+
+    expect(() => schema.string({ validate }).validate('')).toThrowErrorMatchingSnapshot();
```

Comment:
Would you mind adding a test that `hostname` still can use string validation rules?
```js
  test('supports string validation rules', () => {
    expect(() =>
      schema.string({ hostname: true, maxLength: 3 }).validate('www.example.com')
    ). toThrowErrorMatchingSnapshot();
  });
```